component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2389,4 +2408,37 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, Map<String,
                     String.format("Failed to update user claims for user : %s.", authenticatedUser), e);
         }
     }
+
+    private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) throws AuthenticationFailedException {
+
+        String username = authenticatedUser.toFullQualifiedUsername();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        try {
+            UserRealm userRealm = getUserRealm(username);
+            if (userRealm == null) {
+                throw new AuthenticationFailedException("UserRealm is null for user : " + username);
+            }
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("userStoreManager is null for user realm: " + userRealm);
+                }
+                throw new AuthenticationFailedException("userStoreManager is null for user realm: " + userRealm);
+            }
+            Map<String, String> claimValues = userStoreManager
+                    .getUserClaimValues(tenantAwareUsername,
+                            new String[]{EmailOTPAuthenticatorConstants.ACCOUNT_UNLOCK_TIME_CLAIM}, null);
+            if (claimValues.get(EmailOTPAuthenticatorConstants.ACCOUNT_UNLOCK_TIME_CLAIM) == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No value configured for claim: " +
```

Comment:
Shall we add more context into this log?

Reply:
fixed in b87fff7